### ISSUE TITLE
Auto-rotate feature fails for certain images

### ISSUE THREAD

**Issue by Balearica** [naptha/tesseract.js#747]
While setting `rotateAuto: true` generally works as expected, for certain documents the angle is falsely calculated as 0 degrees.  Different images can be experimented with using the [image processing example](https://github.com/naptha/tesseract.js/blob/master/examples/browser/image-processing.html).  An image that the feature does not work for is attached below. 

Notably, this is not a reason to not use `rotateAuto`, as the images I've encountered that fail to rotate correctly are not rotated at all--so end up the same as the input image. 

<img src="https://user-images.githubusercontent.com/100809261/235337991-ac11fdea-822d-406c-aa13-cfc1eb1f5c58.png" width="300">

**Comment by Balearica** [comment]
I added an [auto-rotate benchmark](https://github.com/naptha/tesseract.js/blob/master/benchmarks/browser/auto-rotate-benchmark.html) to assess this feature more robustly.  The benchmark rotates each of our [benchmark images](https://github.com/naptha/tesseract.js/tree/master/benchmarks/data) by 0.2, 0.1, -0.1, and -0.2 radians (0.1 radian is about 6 degrees) and attempts to un-rotate them using the auto-rotate option. 

At the time of writing this (version 4.0.3) auto-rotate worked correctly for 4 of 6 images rotated +/- 0.1 radians, and 0 of 6 images rotated +/- 0.2 radians.  This indicates that the feature struggles with larger angles.  Notably, for the images that were not correctly rotated, 0 rotation was applied--so none of the images got worse by using the `rotateAuto` option.

**Comment by Balearica** [comment]
I updated Tesseract.js and Tesseract.js-core to have improved auto-rotate functionality.  Previously auto-rotate worked correctly for 4 of 6 images rotated +/- 0.1 radians, and 0 of 6 images rotated +/- 0.2 radians.  After the change auto-rotate worked correctly for 6 of 6 images rotated +/- 0.1 radians and 3 of 6 images rotated +/- 0.2 radians. 

The change I made was switching from using the Tesseract "reskew angle" to using the "gradient".  Although these are both estimates of the rotation of the page, the `gradient` is calculated later on using more information (it is calculated using individual lines of text), and appears to be significantly better.

I do not believe further improvements are possible using only statistics already calculated by Tesseract.  In the images where the auto-rotation fails, it looks like the root cause is that Tesseract does not detect text lines to begin with.
